### PR TITLE
feat(core): add support for relative date parsing

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: internal/args/unmarshal.go [hunk 1]
```diff
@@ -61,13 +62,24 @@ var unmarshalFuncs = map[reflect.Type]UnmarshalFunc{
 
 	reflect.TypeOf((*time.Time)(nil)).Elem(): func(value string, dest interface{}) error {
 		// Handle absolute time
-		t, err := time.Parse(time.RFC3339, value)
-		if err != nil {
-			return err
+		absoluteTimeParsed, absoluteErr := time.Parse(time.RFC3339, value)
+		// Handle relative time
+		relativeTimeParsed, relativeErr := tparse.ParseNow(time.RFC3339, value)
```
Comment (jerome-quere) [inline internal/args/unmarshal.go:67]: could the lib parse something like `2020-01-06T12:00:00-3h` ?

Comment (remyleone) [reply internal/args/unmarshal.go:67]: We could but it is not guaranteed to work for all units: https://github.com/karrick/tparse#absoluteduration

File: cmd/scw/testdata/test-all-usage-help-date-usage.golden [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  Absolute time refers to a specific and absolute point in time.
+  CLI uses RFC3339 to parse those time and pass a time.Time go structure to the underlying functions.
+
+	Example: "2006-01-02T15:04:05Z07:00"
+
+- Relative time
+
+  Relative time refers to a time calculated from adding a given duration to the time when a command is launched.
+
+	Example: now+1d4m => current time plus 1 day and 4 minutes
```
Comment (jerome-quere) [inline cmd/scw/testdata/test-all-usage-help-date-usage.golden:18]: wrong example

File: internal/args/errors.go [hunk 3]
```diff
@@ -187,3 +187,19 @@ func missingIndices(index, length int) string {
...
 	return strings.Join(s, ",")
 }
+
+type CannotParseDateError struct {
+	ArgValue               string
+	AbsoluteTimeParseError error
+	RelativeTimeParseError error
+}
+
+func (e *CannotParseDateError) Error() string {
```
Comment (jerome-quere) [inline internal/args/errors.go:197]: I think I would prefer to return a much simple error and handle the pretty error in the core package

File: internal/args/errors.go [hunk 4]
```diff
@@ -187,3 +187,19 @@ func missingIndices(index, length int) string {
...
+	return fmt.Sprintf(`date parsing error: could not parse %s as either an absolute time (RFC3339) nor a relative time (now(+/-)RFC3339)
+
+Try using RFC3339 for absolute time '2006-01-02T15:04:05Z07:00'. You can also have relative time with 'now +/- a duration' ex: 'now+1d-3w4mo+7y6h4m'
+
+AbsoluteTimeParseError: %s
```
Comment (jerome-quere) [inline internal/args/errors.go:202]: IMHO Too verbose

**Review by jerome-quere — CHANGES_REQUESTED** [review]

**Review by kindermoumoute — CHANGES_REQUESTED** [review]

**Review by kindermoumoute — APPROVED** [review]
LGTM

**Review by jerome-quere — DISMISSED** [review]
LGTM

**Review by jerome-quere — APPROVED** [review]

**Pull request merged**